main.c

```diff
@@ -1146,6 +1173,8 @@ int main(int argc, char **argv)
 	tcmulib_close(tcmulib_context);
 err_free_handlers:
 	darray_free(handlers);
+close_fd:
+	close(fd);
```

Comment:
Why for the normal shutdown we do a F_UNLCK and a close, but for the error path we only do a close on the fd?

From the man page it sounds like the same thing happens and the lock is released due to the process exiting. Is that correct? Is it just customary to only do a close in error paths liek this?

Reply:
From man:

* As well as being removed by an explicit F_UNLCK, record locks are automatically released when the process terminates.

* If a process closes any file descriptor referring to a file, then all of the process's locks on that file are released


I just wrote more like sequence, like open, grab lock, release lock, close in the normal shutdown.
And forget to add release lock at error path.
 
But, just closing the fd without F_UNLCK will also release the lock anyway.

@mikechristie I'm not sure what would you prefer:
1. remove F_UNLCK completely and just close(fd) in the both normal & error paths or
2. add F_UNLCK at error path, which I forget before ?

Thanks!

Reply:
Lol, I do not know.

Go for the F_UNLOCK just to make it clear I guess. We seem to do that for some other things so it will at least be consistent.

Reply:
OK - Make-sense :-)